docs/cli/ilab-rag-retrieval.md

```diff
@@ -0,0 +1,264 @@
+# Ingesting and Utilizing RAG Artifacts Design Proposal
+
+**Author**: Daniele Martinoli
+
+**Version**: 0.1
+
+## 1. Introduction
+Purpose of this document is to propose `ilab` CLI changes to extend the InstructLab flow to generate and utilize RAG artifacts
+in the customer workflows. The proposal includes new commands to ingest a local RAG from pre-processed customer documentation, that we assume to be generated from docling output, and updates to the existing commands to utilize these RAG artifacts.
+
+## 2. Use Case
+* Accelerate POCs of fine-tuning LLMs + RAG.
+* Generate artifacts that data scientists can use to build custom solutions in their development environment:
+  * Jupyter Notebooks & Python Scripts
+  * Podman Desktop
+  * Red Hat OpenShift AI
+  * Internal Red Hat CI systems for products or services (e.g., Lightspeed products)
+
+## 3. Proposed Commands
+**Note**: In the context of version 1.4, currently under development, no changes to the command-line interface should be
+allowed. Therefore, we also propose alternative approaches to run the same RAG pipelines using existing `ilab` commands or
+other provided tools.
+
+### 3.1 RAG Ingestion Pipeline Command
+The proposal is to add a `rag` subgroup under the `data` group, with an `ingest` command, like:
+```
+ilab data rag ingest /path/to/docs/folder
```

Comment:
Some thoughts on this:
1. I guess broadly speaking, I was expecting the proposal for how this should be reflected in the command-line interface to come from members of the engine team, e.g., @cdoern .  However, I guess it is fine for us to propose things here and iterate with them.
2. I don't like `rag ingest` here.  I think we want something that describes what we're doing here, which is building an index rather than bringing in the term "RAG" which describes the _feature_ but not really what this specific step is doing.
3. I'm not sure how to respond to the `/path/to/docs/folder` part.  We definitely want some sort of affordance around a flow where you do an `ilab data generate` and then `ilab` just _knows_ where the outputs of that step are rather than you needing to specify it.  However, some other affordance for being able to override that location also makes sense to me.  So maybe if the folder is optional that solves this?
4. We need to figure out how this fits in with the broader refactor being considered in https://github.com/instructlab/dev-docs/pull/155
5. Also, I would like a flow some day where you can just point this step at source documents and it runs docling for you, but that's lower priority than the flow that is more tightly connected with SDG (or at least SDG preprocessing).

Reply:
> I don't like rag ingest here
2. Me neither, but I was waiting for the closure of the discussion on the related command at #148 that, IIUC, should be the preliminary step before running the embedding ingestion. Depending on the selected verb, we can update this proposal accordingly (maybe, like `ilab data index` or `ilab data generate index`?).

> I'm not sure how to respond to the /path/to/docs/folder part
3. Again, this followed the proposal for the other PR, that has both `--input` and `--output` options.

> We definitely want some sort of affordance around a flow where you do an `ilab data generate` and then ilab just knows where the outputs of that step are rather than you needing to specify it

If this is a valid use case, then yes and the parameter will be optional. We have to think carefully of how to auto-detect the json docs in this case, as the `datasets` folder is "versioned" for each `data generate` execution, so I assume the requirement is to pick all the files from the latest `documents-*` subfolder.